code/core/system/proxy-camera/src/ProxyCamera.cpp

```diff
@@ -32,25 +43,99 @@ namespace proxy {
 
 using namespace std;
 using namespace odcore::base;
+using namespace odcore::data;
+using namespace odtools::recorder;
 
 ProxyCamera::ProxyCamera(const int &argc, char **argv)
-    : TimeTriggeredConferenceClientModule(argc, argv, "proxy-camera") {}
+    : TimeTriggeredConferenceClientModule(argc, argv, "proxy-camera"),
+    m_recorder(),
+    m_camera() {}
 
 ProxyCamera::~ProxyCamera() {}
 
-void ProxyCamera::setUp() {}
+void ProxyCamera::setUp() {
+            // This method will be call automatically _before_ running body().
+            if (getFrequency() < 20) {
+                cerr << endl << endl << "proxy-camera: WARNING! Running proxy-camera with a LOW frequency (consequence: data updates are too seldom and will influence your algorithms in a negative manner!) --> suggestions: --freq=20 or higher! Current frequency: " << getFrequency() << " Hz." << endl << endl << endl;
+            }
+
+            // Get configuration data.
+            KeyValueConfiguration kv = getKeyValueConfiguration();
+
+            // Create built-in recorder.
+            const bool useRecorder = kv.getValue<uint32_t>("proxy-camera.useRecorder") == 1;
+            if (useRecorder) {
+                // URL for storing containers.
+                stringstream recordingURL;
+                recordingURL << "file://" << "proxy_camera" << TimeStamp().getYYYYMMDD_HHMMSS() << ".rec";
```

Comment:
@HangYin2015 proxy-fh16 is using the following file naming (https://github.com/chalmers-revere/opendlv.core/blob/master/code/core/system/proxy-fh16/src/ProxyFH16.cpp#L78-L84 and https://github.com/chalmers-revere/opendlv.core/blob/master/code/core/system/proxy-fh16/src/ProxyFH16.cpp#L131-L133): 

CID-123_can_mapped_data_" << YYYYMMDD_HHMMSS << ...

Could you please adjust the file naming from proxy to:

const string CAMERA_NAME = getKeyValueConfiguration().getValue<string>("proxy-camera.camera.name");

"CID-" << getCID() << "_" << getName() << "_" << CAMERA_NAME << "_" << YYYYMMDD_HHMMSS

where YYYYMMDD_HHMMSS is formatted similar to proxy-fh16 (ie without any blanks)?